src/main/kotlin/FilePathGenerator.kt

```diff
@@ -0,0 +1,103 @@
+import java.io.File
+import java.lang.IllegalArgumentException
+
+enum class SupportedExtensions(val ext: String) {
+    WAV("wav"),
+    MP3("mp3"),
+    JPEG("jpeg"),
+    JPG("jpg"),
```

Comment:
JPEG and JPG shouldn't be different enums; they're the same type and our enum class can help normalize this. ext could be a vararg.